From 71e28720b4b9d67f67501329d249c969e92de365 Mon Sep 17 00:00:00 2001
From: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Date: Fri, 9 Mar 2018 11:50:28 +0700
Subject: [PATCH 401/909] iommu/ipmmu-vmsa: Clean up unncessary micro-TLB
 invalidation

commit ccc35934fbde41eada69346706840a5f3f5e8e42 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Since the micro-TLB is automatically invalidated by hardware when:
  - MMU Page Table entry is updated or released
  - ASID is updated
  - MMU register values are updated,
do not need to flush again in software.

Clean up these unncessary invalidation.

Signed-off-by: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Signed-off-by: Kazuya Mizuguchi <kazuya.mizuguchi.ks@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/iommu/ipmmu-vmsa.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/iommu/ipmmu-vmsa.c b/drivers/iommu/ipmmu-vmsa.c
index 8f41e6aa768a..02a52ecc19cb 100644
--- a/drivers/iommu/ipmmu-vmsa.c
+++ b/drivers/iommu/ipmmu-vmsa.c
@@ -464,7 +464,7 @@ static void ipmmu_utlb_enable(struct ipmmu_vmsa_domain *domain,
 	ipmmu_write(mmu, IMUCTR(utlb),
 		    ipmmu_read(mmu, IMUCTR(utlb)) |
 		    IMUCTR_TTSEL_MMU(domain->context_id) |
-		    IMUCTR_FLUSH | IMUCTR_MMUEN);
+		    IMUCTR_MMUEN);
 }
 
 /*
@@ -1346,7 +1346,7 @@ static int ipmmu_utlbs_restore(struct ipmmu_vmsa_device *mmu)
 				    slave_mmu->asids_val[i]);
 			ipmmu_write(slave_mmu,
 				    IMUCTR(slave_fwspec->ids[i]),
-				    (slave_mmu->utlbs_val[i] | IMUCTR_FLUSH));
+				    slave_mmu->utlbs_val[i]);
 			pr_debug("%d: Restore UTLB[%d]: 0x%x, ASID[%d]: %d\n",
 				 i, slave_fwspec->ids[i],
 				 ipmmu_read(slave_mmu,
-- 
2.17.1

